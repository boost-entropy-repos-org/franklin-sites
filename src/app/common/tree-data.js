export const treeData = [{
  label: 'Item 1',
  term: 'item_1',
  items: [{
    label: 'Item 1 a',
    term: 'item_1_a',
  }, {
    label: 'Item 1 b',
    term: 'item_1_b',
    items: [{
      label: 'Item 1 b A',
      term: 'item_1_b_A',
    }, {
      label: 'Item 1 b B',
      term: 'item_1_b_B',
    }]
  }]
}, {
  label: 'Some Item 2',
  term: 'item_2'
}];

function getPathLabel(path, sep = ' / ') {
  return path.map(item => item.label).join(sep);
}

function flattenTreeData(treeData, path = []) {
  let flattened = [];
  for (const node of treeData) {
    let nodePath = [...path, node]
    flattened = [...flattened, {
      'label': getPathLabel(nodePath),
      'value': nodePath
    }];
    if (node.items) {
      const moreFlattened = flattenTreeData(node.items, nodePath);
      if (moreFlattened.length) {
        flattened = [...flattened, ...moreFlattened];
      }
    }
  }
  return flattened;
}

export const flatTreeData = flattenTreeData(treeData);
