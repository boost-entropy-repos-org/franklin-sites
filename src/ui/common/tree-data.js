const treeData = [{
    label: "Item 1",
    members: [{
      label: "Item 1 a",
    }, {
      label: "Item 1 b",
      members: [{
        label: "Item 1b A",
      }, {
        label: "Item 1b B",
      }]
    }]
  },
  {
    label: "Some Item 2"
  }
];

export default treeData;
