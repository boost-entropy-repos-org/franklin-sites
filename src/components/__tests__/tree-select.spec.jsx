import React from 'react';
import renderer from 'react-test-renderer';
import TreeSelect from '../tree-select';
import { treeData } from '../../mock-data/tree-data';

describe('TreeSelect component', () => {
  test('should render', () => {
    const component = renderer
      .create(<TreeSelect data={treeData} onSelect={d => d} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('TreeSelect component with filter', () => {
  test('should render', () => {
    const component = renderer
      .create(<TreeSelect data={treeData} onSelect={d => d} autocomplete />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
