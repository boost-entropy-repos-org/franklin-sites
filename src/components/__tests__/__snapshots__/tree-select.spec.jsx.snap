// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeSelect component should render 1`] = `
<div
  className="dropdown-container"
>
  <button
    className="button dropdown"
    onClick={[Function]}
    type="button"
  >
    Select
  </button>
  <div
    className="dropdown-menu"
  />
</div>
`;

exports[`TreeSelect component with filter should render 1`] = `
<div
  className="dropdown-container"
>
  <button
    className="button dropdown"
    onClick={[Function]}
    type="button"
  >
    Select
  </button>
  <div
    className="dropdown-menu"
  />
</div>
`;
